src/main/java/websocket/WSManager.java

```diff
@@ -0,0 +1,160 @@
+        this.notificationHandlerHashMap = new HashMap<>();
+        this.requestHashMap = new HashMap<>();
+        this.socket = new WSConnection(config);
+    }
+
+    // used for testing
+    WSManager(WSConnection socket) {
+        this.notificationHandlerHashMap = new HashMap<>();
+        this.requestHashMap = new HashMap<>();
+        this.socket = socket;
+    }
+
+    /**
+     * Register an INotificationHandler to receive notifications from the server.
+     *
+     * @param type
+     * @param handler
+     */
+    public void registerNotificationHandler(String type, INotificationHandler handler) {
+        notificationHandlerHashMap.put(type, handler);
+    }
+
+    /**
+     * Deregister the INotificationHandler saved for the given type of notification.
+     *
+     * @param type
+     */
+    public void deregisterNotificationHandler(String type) {
+        notificationHandlerHashMap.remove(type);
+    }
+
+    /**
+     * Send a request over the WebSocket connection.
+     *
+     * @param request
+     */
+    public void sendRequest(Request request) throws ConnectException {
+        if (socket.getState() == WSConnection.State.CREATED || socket.getState() == WSConnection.State.ERROR) {
+            try {
+                socket.connect();
+            } catch (Exception e) {
+                logger.error("WebSocket connection could not connect.");
+                throw new ConnectException("Could not connect to WebSocket.");
+            }
+        }
+        String messageText;
+        try {
+            messageText = mapper.writeValueAsString(request);
+        } catch (JsonProcessingException e) {
+            logger.error("Could not map request to Json string: " + request);
+            return;
+        }
+        socket.enqueueMessage(messageText);
+    }
+
+    @Override
+    public void handleMessage(String message) {
+        ServerMessageWrapper wobject;
+        try {
+            wobject = mapper.readValue(message, ServerMessageWrapper.class);
+        } catch (IOException e) {
+            logger.error("Malformed message from server: " + message);
+            return;
+        }
+        switch (wobject.getType()) {
+            case ServerMessageWrapper.TYPE_NOTIFICATION:
+                handleNotification(wobject);
+                break;
+            case ServerMessageWrapper.TYPE_RESPONSE:
+                handleResponse(wobject);
+                break;
+        }
+    }
+
+    private void handleNotification(ServerMessageWrapper wobject) {
+        Notification an;
+        try {
+            an = mapper.convertValue(wobject.getMessageJson(), Notification.class);
+        } catch (IllegalArgumentException e) {
+            String notificationMessage = wobject.getMessageJson().toString();
+            logger.error("Malformed notification from server: " + notificationMessage);
+            return;
+        }
+
+        String key = an.getResource() + '.' + an.getMethod();
```

Comment:
When keying on the Resource and Method, our Go code didn't end up concatenating a dot. Both ways work just fine, but we should probably decide on a key format that we use across languages.